cmd/node-cache/main.go

```diff
@@ -167,15 +172,19 @@ func (c *cacheApp) parseAndValidateFlags() error {
 		flag.PrintDefaults()
 	}
 
-	flag.StringVar(&c.params.localIP, "localip", "", "ip address to bind dnscache to")
+	flag.StringVar(&c.params.localIPStr, "localip", "", "comma-separated string of ip addresses to bind dnscache to")
 	flag.StringVar(&c.params.interfaceName, "interfacename", "nodelocaldns", "name of the interface to be created")
 	flag.DurationVar(&c.params.interval, "syncinterval", 60, "interval(in seconds) to check for iptables rules")
 	flag.StringVar(&c.params.metricsListenAddress, "metrics-listen-address", "0.0.0.0:9353", "address to serve metrics on")
 	flag.BoolVar(&c.setupIptables, "setupiptables", true, "indicates whether iptables rules should be setup")
 	flag.Parse()
 
-	if net.ParseIP(c.params.localIP) == nil {
-		return fmt.Errorf("Invalid localip specified - %q", c.params.localIP)
+	for _, ipstr := range strings.Split(c.params.localIPStr, ",") {
```

Comment:
Should we error out when no IP is passed in (e.g. empty string)?

Reply:
That case will also enter the loop since strings.Split returns the empty string. We will error out in line 185 below.